include/net/net_pkt.h

```diff
@@ -2241,6 +2241,21 @@ void net_pkt_append_buffer(struct net_pkt *pkt, struct net_buf *buffer);
  */
 size_t net_pkt_available_buffer(struct net_pkt *pkt);
 
+/**
+ * @brief Get available buffer space for payload from a pkt
+ *
+ * Note: Unlike net_pkt_available_buffer(), this will take into account the
+ *       headers space.
+ *
+ * @param pkt   The net_pkt which payload buffer availabality should
+ *              be evaluated
+ * @param proto The IP protocol type (can be 0 for none).
+ *
+ * @return the amount of buffer available for payload
+ */
+size_t net_pkt_available_payload_buffer(struct net_pkt *pkt,
```

Comment:
Better to use just "u16_t proto". As we started to have other protocol types.

Reply:
Not in this case no, here it has to be a valid IPPROTO_* value (or 0).
Usage of u16_t in net_context is because of sock_raw, and net_pkt will never asses any header size on AF_PACKET (or AF_UNSPEC etc...).

Reply:
ok